### ISSUE TITLE
I can not add items to my cart

### ISSUE THREAD

**Issue by olmoszamudio** [segmentio/analytics-magento#79]
Since I've updated your extension for Magento I can not add items to my cart. 

ERR (3): Warning: Missing argument 2 for Segment_Analytics_Helper_Data::{closure}()  in /var/www/clients/client0/web1/web/app/code/community/Segment/Analytics/Helper/Data.php on line 139

Could you tell me any advice to fix this please?

**Comment by benmccormick** [comment]
I'm currently seeing an even worse version of this problem.  Attempting to add a product to a cart in our test environment with the cartridge installed resulted in the following error

```
There has been an error processing your request

product_not_exists

Trace:
#0 /usr/share/nginx/html/magento/app/code/core/Mage/Catalog/Model/Api/Resource.php(123): Mage_Api_Model_Resource_Abstract->_fault('product_not_exi...')
#1 /usr/share/nginx/html/magento/app/code/core/Mage/Catalog/Model/Product/Api.php(131): Mage_Catalog_Model_Api_Resource->_getProduct(Object(Mage_Catalog_Model_Product), NULL, NULL)
#2 /usr/share/nginx/html/magento/app/code/community/Segment/Analytics/Helper/Data.php(95): Mage_Catalog_Model_Product_Api->info(Object(Mage_Catalog_Model_Product))
#3 /usr/share/nginx/html/magento/app/code/community/Segment/Analytics/Model/Controller/Addtocart.php(9): Segment_Analytics_Helper_Data->getNormalizedProductInformation(Object(Mage_Catalog_Model_Product))
#4 /usr/share/nginx/html/magento/app/code/community/Segment/Analytics/Model/Controller/Base.php(11): Segment_Analytics_Model_Controller_Addtocart->getBlock(Object(Segment_Analytics_Block_Template))
#5 /usr/share/nginx/html/magento/app/code/community/Segment/Analytics/Model/Front/Controller.php(85): Segment_Analytics_Model_Controller_Base->dispatch()
#6 /usr/share/nginx/html/magento/app/code/community/Segment/Analytics/Model/Observer.php(32): Segment_Analytics_Model_Front_Controller->getBlocks()
#7 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Model/App.php(1338): Segment_Analytics_Model_Observer->addContainerBlock(Object(Varien_Event_Observer))
#8 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Model/App.php(1317): Mage_Core_Model_App->_callObserverMethod(Object(Segment_Analytics_Model_Observer), 'addContainerBlo...', Object(Varien_Event_Observer))
#9 /usr/share/nginx/html/magento/app/Mage.php(447): Mage_Core_Model_App->dispatchEvent('controller_acti...', Array)
#10 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Controller/Varien/Action.php(384): Mage::dispatchEvent('controller_acti...')
#11 /usr/share/nginx/html/magento/app/code/core/Mage/Cms/Helper/Page.php(137): Mage_Core_Controller_Varien_Action->renderLayout()
#12 /usr/share/nginx/html/magento/app/code/core/Mage/Cms/Helper/Page.php(52): Mage_Cms_Helper_Page->_renderPage(Object(Mage_Cms_IndexController), 'home')
#13 /usr/share/nginx/html/magento/app/code/core/Mage/Cms/controllers/IndexController.php(45): Mage_Cms_Helper_Page->renderPage(Object(Mage_Cms_IndexController), 'home')
#14 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Controller/Varien/Action.php(419): Mage_Cms_IndexController->indexAction()
#15 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(250): Mage_Core_Controller_Varien_Action->dispatch('index')
#16 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Controller/Varien/Front.php(176): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#17 /usr/share/nginx/html/magento/app/code/core/Mage/Core/Model/App.php(354): Mage_Core_Controller_Varien_Front->dispatch()
#18 /usr/share/nginx/html/magento/app/Mage.php(683): Mage_Core_Model_App->run(Array)
#19 /usr/share/nginx/html/magento/index.php(87): Mage::run('default', 'store')
#20 {main}
```

That has persisted past the initial page, and the whole site is currently crashed.

**Comment by hankim813** [comment]
@sperand-io can you look at this?

**Comment by benmccormick** [comment]
@sperand-io @hankim813 was this fixed by 0bb6f8669a99ec204d5145d4f4119b93ffc41727 ?

**Comment by oshipper** [comment]
Anyone care to reply? Was getting ready to update to this new version but seems we should not

**Comment by sperand-io** [comment]
hey folks, my sincerest apologies for this bug and the lack of communication here. this (to the extent I was able to reproduce) was indeed fixed by `0bb6f86` and reflected in the latest release. If not please let me know! @benmccormick @oshipper @olmoszamudio
